python/triton/language/core.py

```diff
@@ -1163,6 +1163,29 @@ def xor_sum(input, axis, _builder=None):
     return semantic.xor_sum(input, axis, _builder)
 
 
+@builtin
+@_add_reduction_docstr("prod")
+def prod(input, axis, _builder=None):
+    axis = _constexpr_to_value(axis)
+    return semantic.prod(input, axis, _builder)
+
+
+@builtin
+@_add_reduction_docstr("argmin2")
+def argmin2(input, axis, _builder=None):
+
+    axis = _constexpr_to_value(axis)
+    n = input.shape[axis]
+    index = arange(0, n, _builder=_builder)
+    new_shape = [constexpr(1)] * len(input.shape)
+    new_shape[axis] = constexpr(n)
+    index = view(index, new_shape, _builder=_builder)
+    index = broadcast_to(index, input.shape, _builder=_builder)
+
+    values, indices = semantic.min_with_index(input, index, axis, _builder)
+    return indices
```

Comment:
This is my strategy for `armin`/`argmax`. Instead of special casing it I just lower it as a reduction over two tensors:
```
  %7 = tt.make_range {end = 256 : i32, start = 0 : i32} : tensor<256xi32>
  %8 = tt.view %7 : (tensor<256xi32>) -> tensor<1x256xi32>
  %9:2 = "tt.generic_reduce"(%6, %8) ({
  ^bb0(%arg4: f32, %arg5: i32, %arg6: f32, %arg7: i32):
    %15 = arith.cmpf olt, %arg4, %arg6 : f32
    %16 = arith.cmpf ogt, %arg4, %arg6 : f32
    %17 = arith.minsi %arg5, %arg7 : i32
    %18 = arith.select %16, %arg7, %17 : i32
    %19 = arith.select %15, %arg5, %18 : i32
    %20 = arith.minf %arg4, %arg6 : f32
    tt.generic_reduce.return %20, %19 : f32, i32
  }) {axis = 1 : i32} : (tensor<1x256xf32>, tensor<1x256xi32>) -> (tensor<1xf32>, tensor<1xi32>)
```

This has some really nice properties. 
1. the reduction code is the same whether you discard the min/max value or not
2. It generalized perfectly to higher numbers of tensors, e.g. the 3 needed for aten.var_mean
3. argmin/argmax specific logic is defined entirely at python level
4. In my limited testing so far, it performs identically